### PR TITLE
Improved packages metadata

### PR DESCRIPTION
**Pull request by LeoColomb**
* Adds the `provides` field
* Improve version constraints
* Cleaner and more tested API

### CODE REVIEW

File: tests/Package/PacakgeTest.php [hunk 1]
```diff
@@ -24,12 +24,12 @@ class PacakgeTest extends TestCase
     protected function setUp(): void
```
Comment (swalkinshaw) [inline tests/Package/PacakgeTest.php:24]: Just noticed this filename is typod: `Pacakge` => `Package`

**Review by swalkinshaw — APPROVED** [review]

**Comment by swalkinshaw** [comment]
@LeoColomb I just added you as a collaborator for this repo ✨

File: tests/Package/PacakgeTest.php (hunk 1 above)
Comment (LeoColomb) [reply tests/Package/PacakgeTest.php:24]: Oh good catch! Thx, fixed

**Comment by LeoColomb** [comment]
Alright, though I'll still ask for reviews on PR 😊

**Pull request merged**